### PR TITLE
[TrueID] Add extractor

### PR DESCRIPTION
**Pull request by MinePlayersPE**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

TODO:
- [x] Support non-Indonesian domains (trueid.ph (?), vn.trueid.net, ~~movie.trueid.net~~)
- [ ] Series extractor? (Delegated for another PR)

### CODE REVIEW

**Comment by MinePlayersPE** [comment]
Ready for review?

File: yt_dlp/extractor/trueid.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+            'release_timestamp': 1593536400,
+            'release_year': 1982,
+            'cast': list,
+            'thumbnail': 'https://cms.dmpcdn.com/movie/2020/09/18/8b6e35c0-f97f-11ea-81fe-c52fc9dd314f_original.png',
+            'upload_date': '20200916',
+            'release_date': '20200630',
+        },
+        'params': {
+            'format': 'bv'
+        }
```
Comment (pukkandan) [inline yt_dlp/extractor/trueid.py]:
```py
        'params': {
            'format': 'bv'
        }
```

is not needed in tests anymore. I have changed tests' default params to handle this automatically

File: yt_dlp/extractor/trueid.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        'url': 'https://trueid.ph/series/l8rvvAw7Jwv8/l8rvvAw7Jwv8/naruto-trailer/',
+        'only_matching': True,
+    }]
+    _CUSTOM_RATINGS = {
+        'PG': 7,
+    }
+
+    def _real_extract(self, url):
+        match = self._match_valid_url(url).groupdict()
+        domain, video_id = match['domain'], match['id']
```
Comment (pukkandan) [inline yt_dlp/extractor/trueid.py:95]:
```suggestion
        domain, video_id = self._match_valid_url(url).group('domain', 'id')
```

File: yt_dlp/extractor/trueid.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        except ExtractorError as e:
+            if isinstance(e.cause, compat_HTTPError):
+                errmsg = self._parse_json(e.cause.read().decode(), video_id)['meta']['message']
+                if 'country' in errmsg:
+                    self.raise_geo_restricted(
+                        errmsg, [initial_data['display_country']] if initial_data.get('display_country') else None, True)
+                else:
+                    self.raise_no_formats(errmsg, video_id=video_id)
+            else:
+                raise e
```
Comment (pukkandan) [inline yt_dlp/extractor/trueid.py:112]:
```suggestion
            if not isinstance(e.cause, compat_HTTPError):
                raise e
            errmsg = self._parse_json(e.cause.read().decode(), video_id)['meta']['message']
            if 'country' in errmsg:
                self.raise_geo_restricted(
                    errmsg, [initial_data['display_country']] if initial_data.get('display_country') else None, True)
            else:
                self.raise_no_formats(errmsg, video_id=video_id)
```

**Review by pukkandan — APPROVED** [review]
otherwise lgtm

File: yt_dlp/extractor/trueid.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+            else:
+                formats = [{'url': stream_url}]
+
+        thumbnails = []
+        for thumb_key in initial_data.get('thumb_list') or {}:
+            if url_or_none(initial_data['thumb_list'][thumb_key]):
+                thumbnails.append({
+                    'id': thumb_key,
+                    'url': url_or_none(initial_data['thumb_list'][thumb_key])
+                })
```
Comment (pukkandan) [inline yt_dlp/extractor/trueid.py:130]:
```suggestion
        thumbnails = [{
            'id': thumb_key,
            'url': url_or_none(thumb_url)
        } for thumb_key, thumb_url in initial_data.get('thumb_list') or {} if url_or_none(thumb_url)]
```

**Pull request merged**